### DISCUSSION TITLE
Lost image meta in v2 to v3 migration

### DISCUSSION THREAD

**Discussion by Ravenna** [statamic/migrator#96] [General]
Moving a pretty good-sized site statamic 2 site to v3. While some parts have been easier, others have been challenging like this one.  For reference I am using s3 for images.

In v2 in my assets.yml file I have:
  filename.jpg:
    caption: 'Caption text here'
    focus: 40-60

In v3 I have:
  - assets/filename.jpg

If I go into CP and add a caption and focal point, the meta saves.  I assume its saving images/.meta/filename.jpg in the s3 bucket.  

The issue is I have hundreds of images that now have no captions, focal points or any meta at all.

What can I do to migrate that meta content into my v3 build?

**Comment by jasonvarga** [comment]
Did you use the migrator?

**Reply by Ravenna** [reply]
@jasonvarga we did.  IS the migrator supposed to handle this?

**Reply by jasonvarga** [reply]
Moving this to the migrator repo, then.

**Comment by jesseleite** [comment]
We do migrate focal points, etc.

Could you upload your `site-migration-log-XXXXXXXXX.yaml` file (the X's are a timestamp for when you ran the migration)? Assuming there are errors and warnings, you should find this at your v3 app root.

**Comment by Ravenna** [comment]
@jesseleite it has a bunch of customer-specific info.  what can I look for?

**Reply by jesseleite** [reply]
In the migration output, or in your site migration log file, you should see a warning something like this...

```
Could not generate asset meta.
Please ensure proper configuration on your [disk] disk in [config/filesystems.php],
Then run `php please migrate:asset-container [handle] --meta-only` to complete meta migration.
```

**Comment by Ravenna** [comment]
yep its there.  ok, I'll dig into this.

**Reply by jesseleite** [reply]
Yep, that's because the migrator didn't have access to your S3 container at the time of migration, so it instructs you to configure your asset container filesystem disk and/or .env credentials, then once you've done that, `php please migrate:asset-container [handle] --meta-only` will migrate the meta for you 👍

**Reply by Ravenna** [reply]
@jesseleite one question: do I need to put back the old yaml file so that the migrator has access to the old meta?

**Reply by jesseleite** [reply]
Your v2 `site` folder, as well as any asset folders, should always be present when running a migrator, yes. If you blew them away, you will need to put them back, since all the migrators reference those files.
